### ISSUE TITLE
Easy way to get filesystem location of CtType..?

### ISSUE THREAD

**Issue by shoeoffhead** [INRIA/spoon#3213]
I would like to get the location of a CtType on the filesystem. This would either be:

1. a .java file
2. a folder on the filesystem
3. a .jar or .zip on the filesystem

The only way to get this information at the moment would be to use my own java.util.Map and populate it with every InputResource as key and the list of files it contains as value.

In my usecase I don't want Spoon to write out sources for me. I just want to do some analysis and then do the writing out myself.

**Comment by MartinWitt** [comment]
Hi,

`CtType<?> foo = ...;
		foo.getPosition().getFile().getAbsolutePath();`

should do the trick.

**Comment by shoeoffhead** [comment]
I already tried everything with SourcePosition and CompilationUnit. But I don't get the filesystem-location of the Jar-Archive that contains the .java file that defines the CtType.

The output of type..getPosition().getFile().getAbsolutePath(); is just:
/mnt/data/workspace/Spoon-Playground/com/jcraft/jsch/IO.java

com/jcraft/jsch/IO.java is not in that directory. I read it in with:

ZipFolder zipfolder = new ZipFolder(new File("./commons-io-2.6-sources.jar"));
launcher.addInputResource(zipfolder);

**Comment by monperrus** [comment]
> In my usecase I don't want Spoon to write out sources for me.

You can use `--output-type nooutput`

**Comment by shoeoffhead** [comment]
The problem is not with the output. I know how to deactivate this with Spoon. I just wanted to say, that I write out the files myself and for that I would need from Spoon the info where every file is located. Diving deeper into Spoon I realize that this probably impossible using the public API.

For example I would like to ask Spoon where is com.apache.commons.IO.java and Spoon should give me the path to the .jar archive containig IO.java.

I did a little workaround, where I put all the files and there location (folder or .jar) into a java.util.Map:

folder = new ZipFolder(path.toFile());
for (SpoonFile f : folder.getAllJavaFiles()) fileLocationMap.put(f.getName(), folder.toString());

Later (after Spoon analysis) I can use fileLocationMap to copy the .java files myself.

In summary: it would be nice if Spoon could give me this information but it is easy to work around the problem. So all is good ;)

**Comment by monperrus** [comment]
thanks for the follow up!
